### PR TITLE
feat(queries): pull ocaml queries from upstream

### PR DESCRIPTION
**Pull request by undu**
This makes the queries compatible with the newest ocaml parser as well.

### CODE REVIEW

File: queries/ocaml/locals.scm [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+; Definitions
+;------------
+
+(value_pattern) @local.definition
+
+; References
+;------------
+
+(value_name) @local.reference
```
Comment (vigoux) [inline queries/ocaml/locals.scm:26]: We don't prefix anything with `local`

File: queries/ocaml/tags.scm [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+; Methods
+;--------
+
+(
+  (comment)? @doc .
+  (method_definition (method_name) @name) @definition.method
+  (#strip! @doc "^\\(\\*\\*?\\s*|\\s\\*\\)$")
+)
+
+(method_invocation (method_name) @name) @reference.call
```
Comment (vigoux) [inline queries/ocaml/tags.scm:99]: We don't have `tags.scm` for now.

File: queries/ocaml/highlights.scm [hunk 3]
```diff
@@ -1,146 +1,148 @@
...
+;------
+
+(
+  (type_constructor) @type.builtin
+  (#match? @type.builtin "^(int|char|bytes|string|float|bool|unit|exn|array|list|option|int32|int64|nativeint|format6|lazy_t)$")
+)
+
+[(class_name) (class_type_name) (type_constructor)] @type
+
+[(constructor_name) (tag)] @tag
```
Comment (vigoux) [inline queries/ocaml/highlights.scm:16]: We don't have `@tag`

File: queries/ocaml/highlights.scm [hunk 4]
```diff
@@ -1,146 +1,148 @@
...
-(attribute_id) @property
-(field_name) @field
+[(module_name) (module_type_name)] @constructor
+
+; Types
+;------
+
+(
+  (type_constructor) @type.builtin
+  (#match? @type.builtin "^(int|char|bytes|string|float|bool|unit|exn|array|list|option|int32|int64|nativeint|format6|lazy_t)$")
```
Comment (vigoux) [inline queries/ocaml/highlights.scm:11]: Maybe you can use the `#contains?` operator here 
```suggestion
  (#contains? @type.builtin "int" "char" "bytes" "string" "float" "bool" "unit" "exn" "array" "list" "option" "int32" "int64" "nativeint" "format6" "lazy_t")
```

**Review by vigoux — CHANGES_REQUESTED** [review]

File: queries/ocaml/locals.scm [hunk 5]
```diff
@@ -1,26 +1,26 @@
...
+(method_definition) @scope
 
-(fun_expression) @local.scope
+(fun_expression) @scope
 
-(match_case) @local.scope
+(match_case) @scope
 
-(for_expression) @local.scope
+(for_expression) @scope
```
Comment (vigoux) [inline queries/ocaml/locals.scm]: You can refactor that in a `[ ]`

**Comment by vigoux** [comment]
Can you rebase on latest master ?

**Review by vigoux — APPROVED** [review]

File: queries/ocaml/highlights.scm (hunk 4 above)
Comment (stsewd) [reply queries/ocaml/highlights.scm]: Contains will match `integer` when you put `int`, isn't an exact match.

**Pull request merged**